src/AlgebraicGeometry/Schemes/ProjectiveAlgebraicSet/Objects/Types.jl

```diff
@@ -1,15 +1,19 @@
 @attributes mutable struct ProjectiveAlgebraicSet{BaseRing<:Field, GradedRingType<:Ring} <: AbsProjectiveAlgebraicSet{BaseRing, GradedRingType}
-  X::ProjectiveScheme{BaseRing,GradedRingType}
-  function ProjectiveAlgebraicSet(X::ProjectiveScheme; check::Bool=true)
-    @check
-    if check
-      is_geometrically_reduced(X) || error("algebraic sets must be geometrically reduced")
+  X::ProjectiveScheme{BaseRing, GradedRingType}
+  Xred::ProjectiveScheme{BaseRing, GradedRingType}
+  function ProjectiveAlgebraicSet(X::ProjectiveScheme{S, T}; is_reduced::Bool=false,
+                                  check::Bool=true) where {S, T}
+    Y = new{S, T}()
+    Y.X = X
+    if is_reduced
+      @check is_geometrically_reduced(X) || error("algebraic sets must be geometrically reduced")
+      Y.Xred = X
     else
       # unlock the scheme methods for geometrically reduced schemes.
       set_attribute!(X, :is_geometrically_reduced, true)
       set_attribute!(X, :is_reduced, true)
```

Comment:
These two lines are altering properties of X, which was the input !!! Is this a typo or is there something I am missing here?

Reply:
You are right, I should add these properties to `Y`.